afs/afs-mapdb-storage/src/main/java/com/powsybl/afs/mapdb/storage/MapDbAppStorage.java

```diff
@@ -772,45 +572,63 @@ public void removeAllTimeSeries(String nodeId) {
     }
 
     @Override
-    public Optional<NodeInfo> getDependency(String nodeId, String name) {
+    public void addDependency(String nodeId, String name, String toNodeId) {
         UUID nodeUuid = checkNodeId(nodeId);
         Objects.requireNonNull(name);
+        UUID toNodeUuid = checkNodeId(toNodeId);
         checkNodeExists(nodeUuid);
-        UUID dependencyNodeUuid = dependencyNodeMap.get(new NamedLink(nodeUuid, name));
-        return Optional.ofNullable(dependencyNodeUuid).map(this::getNodeInfo);
+        checkNodeExists(toNodeUuid);
+        addToList(dependencyNodesMap, nodeUuid, new NamedLink(toNodeUuid, name));
+        addToList(dependencyNodesByNameMap, new NamedLink(nodeUuid, name), toNodeUuid);
+        addToList(backwardDependencyNodesMap, toNodeUuid, nodeUuid);
     }
 
     @Override
-    public void addDependency(String nodeId, String name, String toNodeId) {
+    public Set<NodeInfo> getDependencies(String nodeId, String name) {
         UUID nodeUuid = checkNodeId(nodeId);
         Objects.requireNonNull(name);
-        UUID toNodeUuid = checkNodeId(toNodeId);
         checkNodeExists(nodeUuid);
-        checkNodeExists(toNodeUuid);
-        dependencyNodesMap.put(nodeUuid, dependencyNodesMap.get(nodeUuid).add(toNodeUuid));
-        dependencyNodeMap.put(new NamedLink(nodeUuid, name), toNodeUuid);
-        dependencyNameMap.put(new UnorderedNodeUuidPair(nodeUuid, toNodeUuid), name);
-        backwardDependencyNodesMap.put(toNodeUuid, backwardDependencyNodesMap.get(toNodeUuid).add(nodeUuid));
+        List<UUID> dependencyNodes = dependencyNodesByNameMap.get(new NamedLink(nodeUuid, name));
+        if (dependencyNodes == null) {
+            return Collections.emptySet();
+        }
+        return dependencyNodes.stream().map(this::getNodeInfo).collect(Collectors.toSet());
     }
 
     @Override
-    public List<NodeInfo> getDependencies(String nodeId) {
+    public Set<NodeDependency> getDependencies(String nodeId) {
         UUID nodeUuid = checkNodeId(nodeId);
-        UuidList dependencyNodeUuids = dependencyNodesMap.get(nodeUuid);
-        if (dependencyNodeUuids == null) {
+        List<NamedLink> dependencyNodes = dependencyNodesMap.get(nodeUuid);
+        if (dependencyNodes == null) {
             throw createNodeNotFoundException(nodeUuid);
```

Comment:
The getDependencies(String, String) method return an empty set instead of throwing an exception.